pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -198,6 +197,116 @@ var _ = Describe("podNIC", func() {
 			})
 		})
 	})
+	type stateRetrievalTestCase struct {
+		storedState   *cache.PodIfaceState
+		expectedState cache.PodIfaceState
+	}
+	statePointer := func(state cache.PodIfaceState) *cache.PodIfaceState {
+		return &state
+	}
+	DescribeTable("state is retrieved", func(c stateRetrievalTestCase) {
+		vmi := newVMIMasqueradeInterface("testnamespace", "testVmName", masqueradeCidr, masqueradeIpv6Cidr)
+		podnic, err := newPhase1PodNICWithMocks(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		if c.storedState != nil {
+			podnic.setState(*c.storedState)
+			c.expectedState = *c.storedState
+		}
+		obtainedState, err := podnic.state()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(obtainedState).To(Equal(c.expectedState))
+	},
+		Entry("when there is no pod interface cache should return PodIfaceNetworkPreparationPending", stateRetrievalTestCase{
+			storedState:   nil,
+			expectedState: cache.PodIfaceNetworkPreparationPending,
+		}),
+		Entry("when PodIfaceNetworkPreparationStarted is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationStarted),
+		}),
+		Entry("when PodIfaceNetworkPreparationPending is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationPending),
+		}),
+		Entry("when PodIfaceNetworkPreparationFinished is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationFinished),
+		}),
+	)
+	Context("state transition", func() {
+		When("state is PodIfaceNetworkPreparationPending", func() {
+			var (
+				podnic *podNIC
+			)
+			BeforeEach(func() {
+				var err error
+				vmi := newVMIBridgeInterface("testnamespace", "testVmName")
+				podnic, err = newPhase1PodNICWithMocks(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				pid := 1
+				podnic.launcherPID = &pid
+				podnic.setState(cache.PodIfaceNetworkPreparationPending)
+				mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return("1.2.3.4", "169.254.0.0", nil)
+				mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil)
+				mockPodNetworkConfigurator.EXPECT().DiscoverPodNetworkInterface(podnic.podInterfaceName)
+				mockPodNetworkConfigurator.EXPECT().GenerateDomainIfaceSpec().Return(api.Interface{})
+				mockPodNetworkConfigurator.EXPECT().GenerateNonRecoverableDHCPConfig().Return(&cache.DHCPConfig{})
+			})
+			Context("and prepare networking is fine at phase1 calling", func() {
+				BeforeEach(func() {
+					mockPodNetworkConfigurator.EXPECT().PreparePodNetworkInterface()
+					Expect(podnic.PlugPhase1()).To(Succeed())
+				})
+				It("should transition to PodIfaceNetworkPreparationFinished state", func() {
+					obtainedState, err := podnic.state()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(obtainedState).To(Equal(cache.PodIfaceNetworkPreparationFinished))
```

Comment:
gomega's DSL allows you to inline these, since:

```golang
// Expect wraps an actual value allowing assertions to be made on it:
//    Expect("foo").To(Equal("foo"))
//
// If Expect is passed more than one argument it will pass the *first* argument to the matcher.
// All subsequent arguments will be required to be nil/zero.
//
// This is convenient if you want to make an assertion on a method/function that returns
// a value and an error - a common patter in Go.
//
// For example, given a function with signature:
//    func MyAmazingThing() (int, error)
//
// Then:
//    Expect(MyAmazingThing()).Should(Equal(3))
// Will succeed only if `MyAmazingThing()` returns `(3, nil)`
```

It allows for less code by removing a redundant expectation.